packages/flutter/test/cupertino/sliding_segmented_control_test.dart

```diff
@@ -0,0 +1,1091 @@
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('children'));
+    }
+
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = const Text('Child 1');
+    children[1] = const Text('Child 2');
+
+    try {
+      await tester.pumpWidget(
+        boilerplate(
+          child: CupertinoSlidingSegmentedControl<int>(
+            children: children,
+            controller: null,
+          ),
+        ),
+      );
+      fail('Should not be possible to create segmented control without a controller');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('controller'));
+    }
+
+    try {
+      await tester.pumpWidget(
+        boilerplate(
+          child: CupertinoSlidingSegmentedControl<int>(
+            children: children,
+            controller: ValueNotifier<int>(null),
+            padding: null,
+          ),
+        ),
+      );
+      fail('Should not be possible to create segmented control with null padding');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('padding'));
+    }
+  });
+
+  testWidgets('Need at least 2 children', (WidgetTester tester) async {
+    final Map<int, Widget> children = <int, Widget>{};
+    try {
+      await tester.pumpWidget(
+        boilerplate(
+          child: CupertinoSlidingSegmentedControl<int>(
+            children: children,
+            controller: ValueNotifier<int>(null),
+          ),
+        ),
+      );
+      fail('Should not be possible to create a segmented control with no children');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('children.length'));
+    }
+    try {
+      children[0] = const Text('Child 1');
+
+      await tester.pumpWidget(
+        boilerplate(
+          child: CupertinoSlidingSegmentedControl<int>(
+            children: children,
+            controller: ValueNotifier<int>(null),
+          ),
+        ),
+      );
+      fail('Should not be possible to create a segmented control with just one child');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('children.length'));
+    }
+
+    try {
+      children[1] = const Text('Child 2');
+      children[2] = const Text('Child 3');
+      await tester.pumpWidget(
+        boilerplate(
+          child: CupertinoSlidingSegmentedControl<int>(
+            children: children,
+            controller: ValueNotifier<int>(-1),
+          ),
+        ),
+      );
+      fail('Should not be possible to create a segmented control with a controller pointing to a non-existent child');
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains('value must be either null or one of the keys in the children map'));
+    }
+  });
+
+  testWidgets('Padding works', (WidgetTester tester) async {
+    const Key key = Key('Container');
+
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = const SizedBox(
+      height: double.infinity,
```

Comment:
Maybe include a comment about using height=infinity to ensure that the child consumes all of the available vertical space, like SizedBox.expand() does for both dimensions.

Reply:
turns out I didn't need the `SizedBox` since I'm checking the Rect of the `Opacity` widget